apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -0,0 +1,90 @@
+import { OpenAI } from 'openai';
+
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { QuestionGenerationContextEmbeddingSchema } from '../../lib/db-types.js';
+import { createServerJob } from '../../lib/server-jobs.js';
+
+import { createEmbedding, openAiUserFromLocals, vectorToString } from './contextEmbeddings.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+export async function generateQuestion(
```

Comment:
Let's also avoid using `locals` here in library code. Instead this function should take a `course_id` and `authn_user_id` as arguments. Because we'll be passing multiple string arguments it's probably best to use a single destructured object for arguments, like this:
```
export async function generateQuestion({
  client,
  course_id,
  authn_user_id,
  prompt,
}: {
  client: OpenAI;
  course_id: string | undefined;
  authn_user_id: string;
  prompt: string;
}) {
```
and then you call it like:
```
generateQuestion({
  client,
  course_id: locals.course?.id,
  authn_user_id: locals.authn_user.user_id,
  prompt,
});
```

Reply:
Done in ff0b8fa.